Extension/src/LanguageServer/configurations.ts

```diff
@@ -475,7 +477,7 @@ export class CppProperties {
 
             // Update the compilerPath, cStandard, and cppStandard with the default if they're missing.
             let config: Configuration = this.configurationJson.configurations[this.CurrentConfiguration];
-            if (config.compilerPath === undefined) {
+            if (config.compilerPath === undefined && this.defaultCompilerPath && !config.compileCommands) {
```

Comment:
Why would we turn this off if compileCommands is set?  There are cases where we can't get the compiler info from compile_commands.json because the command begins with a custom script or something.  I believe we would fall back to the guessed compiler in that case and should propagate the chosen compiler back to c_cpp_properties.json